### PR TITLE
new index: switching index from v1 to v2

### PR DESCRIPTION
**Pull request by mhamwala**
Reads the users local appsody `repository.yaml` and switches the v1 index to use the v2 index.

fixes: https://github.com/appsody/appsody/issues/250

### CODE REVIEW

File: cmd/utils.go [hunk 1]
```diff
@@ -1045,3 +1045,22 @@ func checkTime() {
...
+
+	data, err := ioutil.ReadFile(repoFile)
+	if err != nil {
+		Warning.log("Unable to read repository file")
+	}
+
+	replaceURL := bytes.Replace(data, []byte(oldIndexURL), []byte(newIndexURL), -1)
+
+	if err = ioutil.WriteFile(repoFile, replaceURL, 0666); err != nil {
+		fmt.Println(err)
```
Comment (kylegc) [inline cmd/utils.go]: Lets turn this into `Warning.log(err)`

File: cmd/utils.go [hunk 2]
```diff
@@ -1045,3 +1045,22 @@ func checkTime() {
...
+		os.Exit(1)
```
Comment (kylegc) [inline cmd/utils.go]: I think we should remove the os.Exit
If this fails, the user may still be able to use appsody with the old index.

File: cmd/utils.go [hunk 3]
```diff
@@ -1045,3 +1045,22 @@ func checkTime() {
 		}
 	}
 }
+
+func setNewIndexURL() {
```
Comment (kylegc) [inline cmd/utils.go]: Please add some comments to this function to describe its purpose. Especially in this case, I'm looking for something that says this is temporary code and should be removed in the future.

File: cmd/utils.go [hunk 4]
```diff
@@ -1045,3 +1045,22 @@ func checkTime() {
...
+	var newIndexURL = "https://github.com/appsody/stacks/releases/latest/download/incubator-index.yaml"
+
+	data, err := ioutil.ReadFile(repoFile)
+	if err != nil {
+		Warning.log("Unable to read repository file")
+	}
+
+	replaceURL := bytes.Replace(data, []byte(oldIndexURL), []byte(newIndexURL), -1)
+
+	if err = ioutil.WriteFile(repoFile, replaceURL, 0666); err != nil {
```
Comment (kylegc) [inline cmd/utils.go]: Also, I suggest using 0644 to be consistent with the way we write the repo file today.
See https://github.com/appsody/appsody/blob/master/cmd/repo.go#L476

File: cmd/utils.go (hunk 2 above)
Comment (mhamwala) [reply cmd/utils.go]: yep this makes sense!

File: cmd/utils.go (hunk 4 above)
Comment (mhamwala) [reply cmd/utils.go]: change addressed

**Comment by mhamwala** [comment]
Changes addressed :)

**Pull request merged**